### PR TITLE
changes to vacationrequest related schemas

### LINKED ISSUE #43: Modifying vacation request status.
Removing some of vacation request status properties ann removing the endpoints for it

### PR DESCRIPTION
**Pull request by rimz178**

### CODE REVIEW

File: swagger.yaml [hunk 1]
```diff
@@ -335,9 +335,8 @@ paths:
       tags:
         - Interest
       parameters:
-        - name: dealId
-          in: path
-          required: true
+        - name: dealIdgit fetch origin
+git checkout feature-43-modifying-vacation-request-status
```
Comment (dalju005-xamk) [inline swagger.yaml]: What is this ?

File: swagger.yaml [hunk 2]
```diff
@@ -857,10 +856,10 @@ paths:
       tags:
         - VacationRequests
       parameters:
-        - name: personId
+        - name: userId
           required: false
```
Comment (dalju005-xamk) [inline swagger.yaml:861]: Does this false need to be here ? Can it be removed if nothing is required ?

**Review by dalju005-xamk — CHANGES_REQUESTED** [review]

File: swagger.yaml (hunk 2 above)
Comment (naStipe) [reply swagger.yaml:861]: Or actually maybe it should be required...

File: swagger.yaml [hunk 3]
```diff
@@ -335,9 +335,7 @@ paths:
       tags:
         - Interest
       parameters:
-        - name: dealId
-          in: path
-          required: true
+        - name: dealId
```
Comment (naStipe) [inline swagger.yaml]: I suppose that those were not supposed to be deleted

File: swagger.yaml [hunk 4]
```diff
@@ -1648,7 +1474,7 @@ components:
       type: object
       description: Person's single vacation request
       required:
-        - personId
+        - userId
         - days
         - startDateS
```
Comment (naStipe) [inline swagger.yaml]: this renamed into startDate

File: swagger.yaml [hunk 5]
```diff
@@ -1705,25 +1533,12 @@ components:
...
-          type: string
-          format: uuid
-          description: ID of vacation request status in UUID
-          default: null
-        vacationRequestId:
-          type: string
-          format: uuid
-          description: ID of vacation request in UUID
-          default: null
+
```
Comment (naStipe) [inline swagger.yaml]: empty line is not necessary here

File: swagger.yaml [hunk 6]
```diff
@@ -1732,11 +1547,6 @@ components:
           type: string
           format: date-time
           description: Date time of status update
```
Comment (naStipe) [inline swagger.yaml:1576]: Maybe add an empty line between VacationRequestStatus and Flextime

**Review by naStipe — CHANGES_REQUESTED** [review]

File: swagger.yaml (hunk 4 above)
Comment (naStipe) [reply swagger.yaml]: and also createdBy probably should be required as well

File: swagger.yaml [hunk 7]
```diff
@@ -1683,6 +1509,8 @@ components:
           description: Vacation days used for requested vacation
         type:
           $ref: "#/components/schemas/VacationType"
+        status:
```
Comment (naStipe) [inline swagger.yaml:1537]: This should be an array. See "options" property for "Questionnaire" for a reference

File: swagger.yaml [hunk 8]
```diff
@@ -334,8 +334,9 @@ paths:
         Updates string where intrests are stored in Pipedrive
       tags:
         - Interest
-      parameters:
-        - name: dealId
+      parameters:in: path
+          required: true
+        - name: dealId
```
Comment (naStipe) [inline swagger.yaml]: This part is undelined like the error. Some issues with the structure/indexing probably. Try to fix it

File: swagger.yaml [hunk 9]
```diff
@@ -477,12 +478,7 @@ paths:
         "200":
           description: Software entry retrieved successfully
           content:
-            application/json:
-              schema:
-                $ref: "#/components/schemas/SoftwareRegistry"
-        "default":
-          description: Software entry not found
-
+            application/json:and maybe you should make "userId" required for vacationRequests
```
Comment (naStipe) [inline swagger.yaml]: Part of the code is missing and there is a comment from GitHub for some reason

**Review by naStipe — CHANGES_REQUESTED** [review]

File: swagger.yaml (hunk 7 above)
Comment (naStipe) [reply swagger.yaml:1537]: it should look like:
type: array
  items:
    $ref:....

Comment (naStipe) [reply swagger.yaml:1537]: and the status should be an array, not type

**Review by naStipe — CHANGES_REQUESTED** [review]

**Review by naStipe — APPROVED** [review]

**Review by Hanhz2707 — APPROVED** [review]

**Review by dalju005-xamk — APPROVED** [review]

File: swagger.yaml [hunk 10]
```diff
@@ -857,10 +857,10 @@ paths:
       tags:
         - VacationRequests
       parameters:
-        - name: personId
-          required: false
+        - name: userId
+          required: true
```
Comment (Jdallos) [inline swagger.yaml]: We already have the vacations/{id} endpoint to as the find endpoint for vacations. I understand we might want to have userId as a param here if we want to retrieve vacationRequests for specific users- but are we sure it should be required? What is we want to list all vacation requests for all users e.g. in a admin view?

File: swagger.yaml [hunk 11]
```diff
@@ -1703,8 +1531,14 @@ components:
...
           description: Vacation days used for requested vacation
-        type:
+        type: 
           $ref: "#/components/schemas/VacationType"
+          description: type of vacation request
+        status:
+          type: array
+          items:
+            $ref: "#/components/schemas/VacationRequestStatus"
+          description: status of vacation request
```
Comment (Jdallos) [inline swagger.yaml]: Fix indentation
```suggestion
            description: status of vacation request
```

**Review by Jdallos — CHANGES_REQUESTED** [review]

File: swagger.yaml (hunk 10 above)
Comment (naStipe) [reply swagger.yaml]: Don`t we need the userID to "connect" vacation requests to specific users? Even if we get all the vacationRequests, i thought, that we need to know who these requests are for. And if there is a vacationRequest without id sent to/received from the server, doesn`t it indicate an issue? Correct me if I misunderstood.

Comment (naStipe) [reply swagger.yaml]: Nevermind, it is not needed in that place.

Comment (rimz178) [reply swagger.yaml:861]: Should I change the 'required' field to false?

Comment (rimz178) [reply swagger.yaml]: Should I change the 'required' field to false?

Comment (naStipe) [reply swagger.yaml]: yes

**Review by Jdallos — APPROVED** [review]

**Pull request merged**